engine/client/client.go

```diff
@@ -237,21 +243,33 @@ func (c *Client) startEngine(ctx context.Context) (rerr error) {
 		return err
 	}
 
+	c.connector = connector
+
 	ctx, span := Tracer().Start(ctx, "connecting to engine")
 	defer telemetry.End(span, func() error { return rerr })
 
+	logs, slog := slog.SpanLogger(ctx, InstrumentationLibrary, slog.LevelDebug)
```

Comment:
I like this, but - one issue with this is the user (through the frontend) can't configure the logging level.

I hit this before (it's not new) - but ideally it would be really nice if we could somehow defer this decision into the frontend?

Reply:
Totally - just added a `LogLevel` param which gets set by `withEngine`, so now it'll respect `--debug`